### PR TITLE
Flexible verification

### PR DESCRIPTION
**Pull request by fanfilmu**

### CODE REVIEW

File: lib/jwt/token/asymmetric_key_configuration.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module JWT
+  class Token
+    class AsymmetricKeyConfiguration
+      class PublicKeySet
```
Comment (klaudia21) [inline lib/jwt/token/asymmetric_key_configuration.rb:6]: Doesn't this class deserve her own file and tests? 😢

File: lib/jwt/token/verifier.rb [hunk 2]
```diff
@@ -10,7 +10,10 @@ def self.included(base)
 
       module ClassMethods
         def verify(jwt_token, context = nil)
-          decoded = JWT.decode(jwt_token, configuration.secret[:public], true, decode_options)
+          decoded = JWT.decode(jwt_token, nil, true, decode_options) do |header|
+            algorithm_type = JWT::Token::Configuration::ALGORITHMS[header["alg"]]
+            configuration.send(algorithm_type).public_key if algorithm_type
```
Comment (klaudia21) [inline lib/jwt/token/verifier.rb:15]: ~~If `algorithm_type ` is absent we get in line 18 `NoMethodError (undefined method '[]' for nil:NilClass)`. I would rather raise error here that algorithm type is absent.~~

File: lib/jwt/token/asymmetric_key_configuration.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+      attr_accessor :authorized_keys, :private_key
+
+      def initialize(key_class)
+        @key_class = key_class
+      end
+
+      def authorized_keys_file=(file_path)
+        self.authorized_keys =
+          File.read(file_path)
```
Comment (klaudia21) [inline lib/jwt/token/asymmetric_key_configuration.rb:24]: Do we want keep keys in file?

**Review by klaudia21 — APPROVED** [review]
OK!
i

File: lib/jwt/token/verifier.rb (hunk 2 above)
Comment (fanfilmu) [reply lib/jwt/token/verifier.rb:15]: JWT will actually return an error: https://github.com/jwt/ruby-jwt/blob/master/lib/jwt/decode.rb#L63

File: lib/jwt/token/asymmetric_key_configuration.rb (hunk 3 above)
Comment (fanfilmu) [reply lib/jwt/token/asymmetric_key_configuration.rb:24]: It seems like a viable options (these are only public keys anyway).

Think `.ssh/authorized_keys`

**Pull request merged**